src/components/auth/RequireAuth.tsx

```diff
@@ -0,0 +1,17 @@
+import { Navigate, useLocation } from "react-router-dom";
+import useAuth from "hooks/useAuth";
+
+function RequireAuth({ children }: { children: JSX.Element }) {
+  const { email } = useAuth();
```

Comment:
I would not use email as source of truth, I would or add additional parameter - isAuth, when user login, or check that  token exist

Reply:
Ok, I'll change it